File: quiz-api-db/src/Model/Quiz.cs
using System;
using System.Collections.Generic;

namespace Quiz.Api.Model
{
    public class Quiz
    {
        public Quiz()
        {
            Questions = new List<Question>();
        }

        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public bool Active { get; set; }
        public List<Question> Questions { get; set; }

        public void Activate()
        {
            this.Active = true;
        }

        public void Deactivate()
        {
            this.Active = false;
        }
    }
}

File: quiz-api/src/model/quiz.js
import mongoose from 'mongoose';

const optionSchema = new mongoose.Schema({
  id: { type: Number, unique: true },
  name: String,
  correct: Number,
  result: Number
})

const questionSchema = new mongoose.Schema({
  id: { type: Number, unique: true },
  name: String,
  options: [optionSchema]
})

const quizSchema = new mongoose.Schema({
  id: { type: Number, unique: true },
  name: String,
  description: String,
  active: Boolean,
  questions: [questionSchema]
});

export const Quiz = mongoose.model('Quiz', quizSchema);

export function findActiveQuiz(){
  return new Promise((resolve, reject) => {
    Quiz.findOne(
        { active: true},
        (err, quiz) => {
          if (err) reject(err)
          else resolve(quiz);
        }
      );
  });
}
